pkg/metricsexporter/metricsexporter.go

```
@@ -0,0 +1,26 @@
+// Copyright (C) 2022, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package metricsexporter
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// StartMetricsServer starts the metrics endpoint. See StartHTTPsServer for the original implementation from which this function is derived.
+func StartMetricsServer() {
+
+	go wait.Until(func() {
+		http.Handle("/metrics", promhttp.Handler())
+		err := http.ListenAndServe(":9100", nil)
+		if err != nil {
+			zap.S().Errorf("Failed to start metrics server for VMI: %v", err)
```

The typical process for error handling in Golang is to log it at the source and bubble it up to the main function. I think this `StartMetricsServer` function should probably return an error. Although, this might be a bit more difficult with your goroutine. I will look for potential error handling solutions. However, we want to do more than log it and do nothing.

---

gotcha, I will look for solutions as well. I am trying to avoid an observer/producer type of pattern